src/include/souffle/SouffleInterface.h

```diff
@@ -695,21 +727,21 @@ class SouffleProgram {
      * otherwise will add to internalRelations. (a relation could be both input and output at the same time.)
      *
      * @param name the name of the relation (std::string)
-     * @param rel a pointer to the relation (std::string)
+     * @param rel a reference to the relation
      * @param isInput a bool argument, true if the relation is a input relation, else false (bool)
      * @param isOnput a bool argument, true if the relation is a ouput relation, else false (bool)
      */
-    void addRelation(const std::string& name, Relation* rel, bool isInput, bool isOutput) {
-        relationMap[name] = rel;
-        allRelations.push_back(rel);
+    void addRelation(const std::string& name, Relation& rel, bool isInput, bool isOutput) {
```

Comment:
Is this change necessary? It will cause an interface change; what is the rationale?

Reply:
Rationale: Safety - You can't pass in a null reference.
I'll add a deprecated overload for back-compat.

Reply:
Great.